crates/neon/src/types_impl/mod.rs

```diff
@@ -1154,6 +1155,50 @@ impl JsFunction {
     }
 }
 
+pub(crate) unsafe fn call_local<'a, 'b, C: Context<'a>, T, AS>(
+    cx: &mut C,
+    callee: raw::Local,
+    this: Handle<'b, T>,
+    args: AS,
+) -> JsResult<'a, JsValue>
+where
+    T: Value,
+    AS: AsRef<[Handle<'b, JsValue>]>,
+{
+    let (argc, argv) = unsafe { prepare_call(cx, args.as_ref()) }?;
+    let env = cx.env();
+    let mut result: MaybeUninit<raw::Local> = MaybeUninit::zeroed();
+
+    let status = napi::call_function(
+        env.to_raw(),
+        this.to_local(),
+        callee,
+        argc as usize,
+        argv.cast(),
+        result.as_mut_ptr(),
+    );
+
+    match status {
+        sys::Status::InvalidArg => {
+            if !sys::tag::is_function(env.to_raw(), callee) {
+                return cx.throw_error("not a function");
+            }
+            panic!("invalid argument");
+        }
```

Comment:
```suggestion
        sys::Status::InvalidArg if !sys::tag::is_function(env.to_raw(), callee) => {
            return cx.throw_error("not a function");
        }
```

If we move the `is_function` to a match guard, it will fall through to the default arm which asserts against `sys::Status::Ok`. This is a bit more consistent with [other asserts against `sys::Status`](https://github.com/neon-bindings/neon/blob/63551b18cb509a30598c06b9ec0b983a76fc687b/crates/neon/src/types_impl/extract/try_from_js.rs#L84-L88).